File: dotNet - POCs/PagarMe.Net.PoCs/Entities/PoCTransactionCreditCard.cs
using Newtonsoft.Json;
using System;
using System.Runtime;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using PagarMe;
using PagarMe.Base;
using PagarMePoCs.Net.Interfaces;

namespace PagarMePoCs.Net.Entities
{
	public class PoCTransactionCreditCard : IPoC
	{
		Transaction model;

		public PoCTransactionCreditCard ()
		{
		}

		public String Title
        { get { return "Criar Transaction Credit Card"; } }

		public void Create ()
		{
			try {
				model = new Transaction () {
					Amount = 3100,
					PaymentMethod = PaymentMethod.CreditCard,
					CardNumber = "4242424242424242",
					CardHolderName = "Teste PagarMe",
					CardExpirationDate = "0921",
					CardCvv = "123",
					Customer = new Customer () {
						Name = "Teste PagarMe",
						DocumentNumber = "43591017833",
						DocumentType = DocumentType.Cpf,
						Email = "teste@pagar.me",
						Address = new Address () {
							Zipcode = "13223030",
							Neighborhood = "Jardim Paulistano",
							Street = "Av. Brigadeiro Faria Lima",
							StreetNumber = "1811"
						},
						Phone = new Phone () {
							Ddd = "11",
							Number = "12345678"
						}
					},
					PostbackUrl = "http://www.aledsz.com.br/validateRequest.php",
					Metadata = new AbstractModel (PagarMeService.GetDefaultService ()) {
						["idProduto" ] = "13933139"
					}
				};

				model.Save ();
			} catch (PagarMeException ex) {
				throw ex;
			}
		}

		public Model GetModel ()
		{
			return model;
		}
	}
}

File: ruby - POCs/PocTransactionCreditCard.rb
require 'pagarme'

class PocTransactionCreditCard


    def initialize()
    		@card_hash = "card_hash recebido no callback de sucesso Checkout ou lib"
	end


	def createTransaction()

		PagarMe.api_key = "SUA_API"
		transaction  = PagarMe::Transaction.new({

         amount: 10000,
         installments: 12,
         payment_method: "boleto",
         # card_hash: @card_hash,
         postback_url: "http://requestb.in/10taynn1",
         customer: {
           name: "John Smith",
           email: "john@smithassociates.com",
           document_number: "03221744307",
           address: {
             street: "The one on the corner",
             street_number: "034",
             neighborhood: "Suburb",
             city: "Sunderland",
             state: "Sundy",
             zipcode: "09720200"
           },
           phone: {
             ddd: "11",
             number: "15514567"
           }
         },
         metadata: {
           invoice: String.new("01625342001470189905")
         }

    })

            
			puts "------------AFTER CHARGE-------------------"
			transaction.charge
			puts transaction.status
			transaction.status = "paid"
      transaction.save 
      puts transaction.status 
	end 
end


if __FILE__ == $0

   tr = PocTransactionCreditCard.new
   
   tr.createTransaction

end


